### ISSUE TITLE
Can't import the `stardoc` module via bzlmod

### ISSUE THREAD

**Issue by aaronmondal** [bazelbuild/stardoc#156]
I'm getting this error when attempting to use `bazel_dep(name="stardoc", version="0.5.4")` in `MODULE.bazel` with bazel 6.2.0:

```
ERROR: /home/aaron/.cache/bazel/_bazel_aaron/032461956725678fc967949a2a82a277/external/rules_java~5.5.0~toolchains~remote_java_tools/BUILD:372:11: error loading package '@rul
es_java~5.5.0~toolchains~remote_java_tools//java_tools/zlib': Unable to find package for @[unknown repo 'rules_license' requested from @rules_java~5.5.0~toolchains~remote_jav
a_tools]//rules:license.bzl: The repository '@[unknown repo 'rules_license' requested from @rules_java~5.5.0~toolchains~remote_java_tools]' could not be resolved: No reposito
ry visible as '@rules_license' from repository '@rules_java~5.5.0~toolchains~remote_java_tools'. and referenced by '@rules_java~5.5.0~toolchains~remote_java_tools//:singlejar
_cc_bin_main'
```

Looks like `rules_java` can't find `rules_license` which is indeed missing from the `MODULE.bazel` for `rules_java`.

cc @fmeum

**Comment by aaronmondal** [comment]
@fmeum @tetromino Could there be some bug or missing test in the current workflows? I think stardoc CI should've picked this up. The `0.5.6` release still breaks with

```
ERROR: /home/aaron/.cache/bazel/_bazel_aaron/032461956725678fc967949a2a82a277/external/rules_java~6.0.0~toolchains~remote_java_tools/BUILD:372:11: error loading package '@rul
es_java~6.0.0~toolchains~remote_java_tools//java_tools/zlib': Unable to find package for @[unknown repo 'rules_license' requested from @rules_java~6.0.0~toolchains~remote_jav
a_tools]//rules:license.bzl: The repository '@[unknown repo 'rules_license' requested from @rules_java~6.0.0~toolchains~remote_java_tools]' could not be resolved: No reposito
ry visible as '@rules_license' from repository '@rules_java~6.0.0~toolchains~remote_java_tools'. and referenced by '@rules_java~6.0.0~toolchains~remote_java_tools//:singlejar
_cc_bin_main'
```

**Comment by fmeum** [comment]
@aaronmondal If you follow the chain of errors, you can see that `rules_java` is lacking a dep here, not `stardoc`. My fix at https://github.com/bazelbuild/rules_java/pull/111 has been reviewed, but not merged or released yet.

CC @hvadehra

**Comment by aaronmondal** [comment]
What confuses me is that the stardoc CI didn't catch this. I mean yes the issue isn't caused by stardoc itself, but shouldn't all commits starting at https://github.com/bazelbuild/stardoc/commit/9c834aec65ec45f7e9f8fd9835800b5f0d9d2343 be red?

**Comment by tetromino** [comment]
I cannot reproduce the problem with bazel-6.2.1

```shell
$ cat MODULE.bazel
bazel_dep(name = "stardoc", version = "0.5.6", repo_name = "io_bazel_stardoc")
$ cat BUILD
load("@io_bazel_stardoc//stardoc:stardoc.bzl", "stardoc")

stardoc(
    name = "test_doc",
    input = "test.bzl",
    out = "test_doc.md",
)
$ cat WORKSPACE
# empty
$ bazel build //:test_doc --enable_bzlmod
INFO: Analyzed target //:test_doc (73 packages loaded, 1110 targets configured).
INFO: Found 1 target...
Target //:test_doc up-to-date:
  bazel-bin/test_doc.md
INFO: Elapsed time: 9.682s, Critical Path: 3.92s
INFO: 15 processes: 9 internal, 4 darwin-sandbox, 2 worker.
INFO: Build completed successfully, 15 total actions
$
```

So my first suspicion is that the error you are seeing is coming a different usage of rules_java in your repo.

Can you provide exact instructions to reproduce the error you are seeing?

**Comment by aaronmondal** [comment]
I got this error when building the docs for rules_ll [here](https://github.com/eomii/rules_ll/blob/66c1808da3daefc919d49ab9eef4f6593b19261a/ll/BUILD.bazel#L43-L55). This error only occurred when building the documentation and only when using bzlmod. I think there was some duplicate dependency thing going on. The docs build file also loaded skylib and transitively loaded a bunch of other rulesets. One of those dependencies broke stardoc. Probably because some dep overrides some other dep which then breaks the java import in stardoc.

Now that rules_java `6.1.0` is out with @fmeum's fix, stardoc works as long as one manually adds

```python
bazel_dep(name = "rules_java", version = "6.1.0")
```

to `MODULE.bazel`. I think Bazel should pick up that version automatically, but won't for whatever reason. Looks like this was caused by some inter-dependency issue in rules_ll.

Since things work with the latest versions I think it's fine to close this issue. Thanks @tetromino for looking into this and thanks @fmeum for the upstream fix ❤️
